src/Microsoft.AspNet.Hosting/Internal/WebApplication.cs

```diff
@@ -83,29 +81,43 @@ public IServiceProvider ApplicationServices
             }
         }
 
-        public virtual IApplication Start()
+        public IFeatureCollection ServerFeatures
         {
-            var application = BuildApplication();
+            get
+            {
+                return Server?.Features;
+            }
+        }
 
-            var logger = _applicationServices.GetRequiredService<ILogger<HostingEngine>>();
+        public void Initialize()
+        {
+            if (_application == null)
+            {
+                _application = BuildApplication();
+            }
+        }
+
+        public virtual IDisposable Start()
```

Comment:
WebApplication can be created directly so Start needs to call Initialize to make sure it's happened.